### PR TITLE
Issue1630 add process

### LINKED ISSUE #1630: add working process of model development
This issue intends to add working process of model development.

### PR DESCRIPTION
**Pull request by JayHuLBL**
This closes #1630.
- [x] added working process to `development` section of user guide.

### CODE REVIEW

File: Buildings/Resources/Documentation/userGuide/source/development.rst [hunk 1]
```diff
@@ -102,3 +102,38 @@ the pressure drop of the heating coil to zero, and
...
+
+Working process
+---------------
+
+Followings are the working process of contributing.
+
+1. create issue to describe the potential problem or what change should be done
+   to the library. This will generate a issue number (``##``). A good
+   practice is that each issue should address only one problem.
+2. create development branch,
```
Comment (AntoineGautier) [inline Buildings/Resources/Documentation/userGuide/source/development.rst]: - Need to precise where the branch is created: locally or remotely. If locally, need to clone or pull before.
- I would use "feature" branch to comply with git official naming, see [https://www.atlassian.com/git/tutorials/comparing-workflows/feature-branch-workflow](https://www.atlassian.com/git/tutorials/comparing-workflows/feature-branch-workflow)

File: Buildings/Resources/Documentation/userGuide/source/development.rst [hunk 2]
```diff
@@ -102,3 +102,38 @@ the pressure drop of the heating coil to zero, and
...
+
+  - if the issue is not part of a large development, the development branch
+    should be named as ``issue(##)_(change)`` and should be based on the ``master``
+    branch.
+  - if the issue is part of a large development and the development has a
+    staging branch like ``issue(#)_(development)``, the development branch should
+    be named as ``issue(#)_(##)_(change)`` and should be based on the staging
```
Comment (AntoineGautier) [inline Buildings/Resources/Documentation/userGuide/source/development.rst]: I would replace everywhere `development` by `staging` and `change` by `feature`.

File: Buildings/Resources/Documentation/userGuide/source/development.rst [hunk 3]
```diff
@@ -102,3 +102,38 @@ the pressure drop of the heating coil to zero, and
...
+    branch.
+
+3. address the issue,
+
+  - fork ``Buildings`` library repository to your account, or pull the development
+    branch to your own ``Buildings`` library repository,
+  - make the change from the development branch,
```
Comment (AntoineGautier) [inline Buildings/Resources/Documentation/userGuide/source/development.rst]: Pull upstream staging branch from central repo before PR.

File: Buildings/Resources/Documentation/userGuide/source/development.rst [hunk 4]
```diff
@@ -102,3 +102,38 @@ the pressure drop of the heating coil to zero, and
...
+  - submit a pull request to merge the changes in the development branch of
+    your own ``Buildings`` library repository to the same development branch of
+    ``Buildings`` library repository,
+  - merge the pull request.
```
Comment (AntoineGautier) [inline Buildings/Resources/Documentation/userGuide/source/development.rst]: Here it's Michael who eventually would merge.

File: Buildings/Resources/Documentation/userGuide/source/development.rst [hunk 5]
```diff
@@ -102,3 +102,38 @@ the pressure drop of the heating coil to zero, and
...
+
+4. close the issue in ``Buildings`` library repository,
+
+  - submit another pull request to merge the development branch to
+    ``master`` branch,
+  - merge the pull request,
```
Comment (AntoineGautier) [inline Buildings/Resources/Documentation/userGuide/source/development.rst]: Here it is for the project manager who will eventually merge the staging branch: do we need to describe that process?

**Comment by JayHuLBL** [comment]
Close this PR as it will be addressed in [wiki](https://github.com/lbl-srg/modelica-buildings/wiki).

**Pull request closed without merging**